server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -30,21 +43,80 @@ class ObjectService(googleStorageService: GoogleStorageService[IO]) extends Http
       } yield resp
   }
 
-  def localize(req: Localize): IO[Unit] = ???
+  def localize(req: Localize): IO[Unit] = {
+    val res = Stream.emits(req.entries).map {
+      entry =>
+        googleStorageService.getObject(entry.bucketNameAndObjectName.bucketName, entry.bucketNameAndObjectName.blobName, None) //get file from google
+          .through(io.file.writeAll(Paths.get(entry.localObjectPath.asString), blockingEc)) //write file to local disk
+    }.parJoin(10)
+
+    res.compile.drain
+  }
 
-  def checkMeta(req: GetMetadataRequest): IO[Unit] = ???
+  def checkMetadata(req: GetMetadataRequest): IO[MetadataResponse] =
+    for {
+      traceId <- IO(TraceId(randomUUID()))
+      storageLinks <- readJsonFileToA[IO, List[StorageLink]](pathToStorageLinks).compile.lastOrError
+      storageLink = storageLinks.find(_.localBaseDirectory == req.localObjectPath)
+      res <- storageLink.fold[IO[MetadataResponse]](IO.raiseError(NotFoundException(s"No storage link found for ${req.localObjectPath}"))) {
+        sl =>
+          for {
+            bucketAndObject <- IO.fromEither(JsonCodec.parseGsDirectory(sl.cloudStorageDirectory.renderString).leftMap(e => InternalException(e)))
+            metadata <- googleStorageService.getObjectMetadata(bucketAndObject.bucketName, bucketAndObject.blobName, Some(traceId)).compile.lastOrError
+            result <- metadata match {
+                        case google2.GetMetadataResponse.NotFound => IO.raiseError(NotFoundException(s"Object(${bucketAndObject}) not found in Google storage"))
+                        case google2.GetMetadataResponse.Metadata(crc32, userDefinedMetadata) =>
+                          for {
+                            isLinkedString <- IO.fromEither(userDefinedMetadata.get("isLinked").toRight(NotFoundException("isLinked key missing from object metadata ")))
```

Comment:
isLinked is just supposed to mean whether a there's a matching directory, i.e. set this to false at line 61 and return, rather than throwing an erro